src/components/TokenTotalLineItem.tsx

```diff
@@ -4,15 +4,17 @@ import { Trans, useTranslation } from 'react-i18next'
 import { StyleSheet, Text } from 'react-native'
```

Comment:
Note that this file is the only one that's not "fully" refactored, since it is used in many places outside the scope of this PR. Option 1 [here](https://www.figma.com/file/iiPCJG2cA18ysZrLAZI80d/Multichain-refactor-strategies?type=whiteboard&node-id=0-1&t=PCFrhpA7ruyapZgH-0) would be too much scope, and this doesn't fit neatly into options 2 or 3 -- perhaps this is a fourth option, which is "make an existing thing backwards compatible"? Though you could perhaps consider this to be a variant of option 3, since it remains DRY by not duplicating the component outright.

Reply:
@jophish to me this looks like a pretty standard case where we would want option 3. How about creating a new TokenTotalLineItem that uses tokenId to identify tokens, and a LegacyTokenTotalLineItem (or "byAddress" version, whatever) that wraps it by getting token info from the token address, then getting the token id from there?

Reply:
the effort here to make this component take either a token address and a tokenId here seems to have some downsides
- practically speaking, there is no easy way for consumers of this component to see that tokenId is the "preferred" way of using it. There isn't even a jsdoc we can use to show it's deprecated. We would rely on people looking into the actual implementation and seeing your todo there.
- this is going to sound kind of fuzzy, but it's how I think about it so here goes. at a high level, the function now serves two purposes and has two fairly distinct execution pathways depending on the inputs provided. to keep code simple, it's better to keep functions single-purpose

Reply:
Yeah -- I think I agree. The change to split this into a `Legacy` component that wraps the new `tokenId`-based one was actually quite small; I should have gone with it from the start I think. Being able to `@deprecate` the old components is actually pretty valuable I think, since allows us to quickly see at a glance where in the app we still need to update.